client/src/components/DisplayAllCampaigns.tsx

```diff
@@ -0,0 +1,34 @@
+import { Button, Flex } from '@radix-ui/themes';
+import { useEffect, useState } from 'react';
+import axios from 'axios';
+import { useNavigate } from 'react-router-dom';
+import { settlementApi } from '../service/api';
+import { TypeServerSettlement } from '../../../SettlementTypes';
+
+const DisplayAllCampaigns = () => {
+    const navigate = useNavigate();
+    const [allCampaigns, setAllCampaigns] = useState<TypeServerSettlement[]>([]);
+
+    useEffect(() => {
+        axios
```

Comment:
don't use `then` instead always use async/await.
For useEffects in particular, you need to get a little fancy:
```
(async () => {
  await someMyThing();
})()
```
Since you're not allowed to return a promise form a useEffect.

Reply:
updated as so:
` useEffect(() => {
        try {
            (async () => {
                const res = await axios.get(`${settlementApi}/get`);
                console.log(res.data);
                setAllCampaigns(res.data.settlements);
            })();
        } catch (err) {
            console.error(err);
        }
    }, []);
`